dask/dataframe/reshape.py

```diff
@@ -59,3 +60,76 @@ def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False,
                           prefix_sep=prefix_sep, dummy_na=dummy_na,
                           columns=columns, sparse=sparse,
                           drop_first=drop_first)
+
+
+def pivot_table(df, index=None, columns=None,
+                values=None, aggfunc='mean'):
+    """
+    Create a spreadsheet-style pivot table as a DataFrame. Target ``columns``
+    must have category dtype to infer result's ``columns``.
+    ``index``, ``columns``, ``values`` and ``aggfunc`` must be all scalar.
+
+    Parameters
+    ----------
+    data : DataFrame
+    values : scalar
+        column to aggregate
+    index : scalar
+        column to be index
+    columns : scalar
+        column to be columns
+    aggfunc : {'mean', 'sum', 'count'}, default 'mean'
+
+    Returns
+    -------
+    table : DataFrame
+    """
+
+    if not is_scalar(index) or index is None:
+        raise ValueError("'index' must be a scalar")
+    if not is_scalar(columns) or columns is None:
+        raise ValueError("'columns' must be a scalar")
+    if not is_categorical_dtype(df[columns]):
+        raise ValueError("'columns' must be category dtype")
+    if not is_scalar(values) or values is None:
+        raise ValueError("'values' must be a scalar")
+    if not is_scalar(aggfunc):
+        raise ValueError("'aggfunc' must be a scalar")
```

Comment:
Updated the error message for ``aggfunc``.

Should I also change other errors, like ``'values' must be a single column name used as 'values'``? (lmk if there is better message)